### PR TITLE
[Connector API] Implement update filtering action

### PR DESCRIPTION
**Pull request by jedrazb**
## Changes

- Implement `PUT _connector/{connector_id}/_filtering` endpoint
- This endpoint expects body with a new connector filtering definition, it overwrites the existing `filtering` field in a given connector doc

## Verification
- Add unit tests for index service
- Add BWC tests
- Add YML tests

Also, verified new endpoint manually.

### CODE REVIEW

File: x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/332_connector_update_filtering.yml [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+  - match: { filtering.0.active.rules.0.id: "RULE-ACTIVE-0" }
+  - match: { filtering.0.draft.rules.0.id: "RULE-DRAFT-0" }
+
+  - match: { filtering.1.domain: TEST }
+  - match: { filtering.1.active.advanced_snippet.created_at: "2021-05-25T12:30:00.000Z" }
+  - match: { filtering.1.active.rules.0.id: "RULE-ACTIVE-1" }
+  - match: { filtering.1.draft.rules.0.id: "RULE-DRAFT-1" }
+
+---
+"Update Connector Filtering - 404 status code returned when connector doesn't exist":
```
Comment (carlosdelest) [inline x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/332_connector_update_filtering.yml:110]: Nit: Maybe we can make this update shorter, so it's easier to read?

**Review by carlosdelest — APPROVED** [review]

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            return connectorId;
+        }
+
+        public List<ConnectorFiltering> getFiltering() {
+            return filtering;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
```
Comment (kderusso) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java:71]: Should we validate that `connectorId` is non null/empty here?

**Review by kderusso — APPROVED** [review]
LGTM!

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java (hunk 2 above)
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java:71]: Added :)

**Pull request merged**